### PR TITLE
fix(#956): handle nested environment

### PR DESCRIPTION
**Pull request by Zomzog**
# Description

This PR aims to solve #956  by recursively resolving environment variables. 

I've blocked it to 10 levels to avoid an infinite loop.

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Screenshot
Tested with this env
Screenshot from 2023-12-03 17-11-57

### CODE REVIEW

File: packages/bruno-electron/tests/network/interpolate-env-vars.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const interpolateEnvVars = require('../../src/ipc/network/interpolate-env-vars');
+
+describe('interpolate wiht nested env vars', () => {
```
Comment (bbbco) [inline packages/bruno-electron/tests/network/interpolate-env-vars.spec.js]: `wiht` => `with`

Comment (Zomzog) [reply packages/bruno-electron/tests/network/interpolate-env-vars.spec.js]: Fixed thanks

**Comment by Nahuel92** [comment]
Hi @helloanoop. Thanks for all your effort and time in making this awesome tool.
Is there any chance for this PR to be merged anytime soon? I would be great if we could use recursive variables in Bruno.

**Comment by bciesielczyk** [comment]
That would be very nice to have

**Comment by pleguen** [comment]
I would love to have this feature, I'm unable to switch to Bruno due to that as working on complex APIs requires to re-use many variables across envs.

I've tried to rebase the PR, however it's scoped to Requests and Env editor, it's missing implementation for `bru.js`. (meaning it does not work inside scripts if you use `getEnvVar` ).

**Comment by sanjai0py** [comment]
Hey @Zomzog, thank you for taking the time to implement this feature. There was another PR #2234 that closely resembles what we needed, so we are merging that! 

Closing the PR.

**Pull request closed without merging**